bokehjs/src/lib/models/expressions/cumsum.ts

```diff
@@ -0,0 +1,49 @@
+import {ColumnarDataSource} from "../sources/columnar_data_source"
+import {Expression} from "./expression"
+import {Arrayable} from "core/types"
+import * as p from "core/properties"
+
+export namespace CumSum {
+  export interface Attrs extends Expression.Attrs {
+    field: string
+    include_zero: boolean
+  }
+
+  export interface Props extends Expression.Props {}
+}
+
+export interface CumSum extends CumSum.Attrs {}
+
+export class CumSum extends Expression {
+
+  properties: CumSum.Props
+
+  constructor(attrs?: Partial<CumSum.Attrs>) {
+    super(attrs)
+  }
+
+  static initClass(): void {
+    this.prototype.type = "CumSum"
+
+    this.define({
+      field:        [ p.String         ],
+      include_zero: [ p.Boolean, false ],
+    })
+  }
+
+  _v_compute(source: ColumnarDataSource): Arrayable<number> {
+    const result = new Float64Array(source.get_length() || 0)
+    const col = source.data[this.field]
+    const offset = this.include_zero ? 1 : 0
+    if (this.include_zero)
+        result[0] = 0
+    else
+        result[0] = col[0]
+    debugger
```

Comment:
^^^

Reply:
We should really just make a code quality check that looks for this

Reply:
Sure, there's a built-in rule for this (https://palantir.github.io/tslint/rules/no-debugger/).

Reply:
fixed in https://github.com/bokeh/bokeh/pull/7961/commits/c59175147ff6c61c95692f1ac13223b61a7161b6

Reply:
If your are at fixing things, you can add `"no-debugger": true` to `bokehjs/tslint.json`.

Reply:
I could try to add it to `quality` but it doesn't currently actually fail when there is a violation:
```
~/work/bokeh/bokehjs bryanv/cumsum_transform*
(base) ❯ node make tslint
[13:32:35] Starting 'tslint:lib'...
[13:32:36] /Users/bryanv/work/bokeh/bokehjs/src/lib/models/expressions/cumsum.ts:35:5
[13:32:36] ERROR: 35:5  no-debugger  Use of debugger statements is forbidden
[13:32:36] Finished 'tslint:lib' after 1.34 s
[13:32:36] Starting 'tslint:test'...
[13:32:36] Finished 'tslint:test' after 116 ms
[13:32:36] Starting 'tslint:examples'...
[13:32:36] Finished 'tslint:examples' after 32 ms
[13:32:36] Finished 'tslint'

~/work/bokeh/bokehjs bryanv/cumsum_transform*
(base) ❯ echo $?
0
```

Reply:
It a part of `codebase` job on travis ci. Otherwise you have to run `node make tslint`. Perhaps we could make it part of `quality` tests at some point.

Reply:
>  but it doesn't currently actually fail when there is a violation:

It was supposed to get `--emit-error`, the same as `scripts:compile` and `test:compile` have. Must have forgotten about this or maybe there is a regression since then new build is in place.

Reply:
OK I added it, tho as I said it's not clear to me that a linter failure results in a test failure since the exit always seems to be 0

Reply:
I tried adding `--emit-error` locally, that still did not result in a nonzero exit code when a lint violation was present

Reply:
In fact this was done on purpose, because initially tslint reported a lot of errors and it was tedious dealing with this. Now all were resolved, so failing `tslint` build task on error should be implemented. I will do it with the next round of build improvements.